src/sessionid.js

```diff
@@ -62,29 +61,41 @@ export class SessionIdManager {
         this._setSessionId(null, null)
     }
 
-    getSessionAndWindowId(timestamp = null, recordingEvent = false) {
-        // Some recording events are triggered by non-user events (e.g. "X minutes ago" text updating on the screen).
-        // We don't want to update the session and window ids in these cases. These events are designated by event
-        // type -> incremental update, and source -> mutation.
-        const isUserInteraction = !(
-            recordingEvent &&
-            recordingEvent.type === INCREMENTAL_SNAPSHOT_EVENT_TYPE &&
-            recordingEvent.data?.source === MUTATION_SOURCE_TYPE
-        )
-
+    /*
+     * This function returns the current sessionId and windowId. It should be used to
+     * access these values over directly calling `._sessionId` or `._windowId`. In addition
+     * to returning the sessionId and windowId, this function also manages cycling the
+     * sessionId and windowId when appropriate by doing the following:
+     *
+     * 1. If the sessionId or windowId is not set, it will generate a new one and store it.
+     * 2. If the shouldExtendExistingSessionOrTriggerNewOne param is set, it will:
+     *    a. Check if it has been > SESSION_CHANGE_THRESHOLD since the last call with this flag set.
+     *       If so, it will generate a new sessionId and store it.
+     *    b. Update the timestamp stored with the sessionId to ensure the current session is extended
+     *       for the appropriate amount of time.
+     *
+     * @param {boolean} shouldExtendExistingSessionOrTriggerNewOne (optional) Defaults to True. Should be set to False when the call to the function should be read only (e.g. being called for non-user generated events)
+     * @param {Number} timestamp (optional) Defaults to the current time. The timestamp to be stored with the sessionId (used when determining if a new sessionId should be generated)
+     */
+    getSessionAndWindowId(shouldExtendExistingSessionOrTriggerNewOne = true, timestamp = null) {
```

Comment:
Much nicer!

Regarding parameter name, one reason why this isn't nice is:

If I come to this function definition and try setting `shouldExtendExistingSessionOrTriggerNewOne = True`, does that mean, just looking at the variable, whether I'll extend existing session, or trigger a new one? (well, maybe it can do either, which is probably where the name is coming from, and the reverse is what we're interested in sometimes, i.e. don't do any modification to existing session and window IDs, in contrast to the default)

So, I propose renaming this to something like: `readOnly = False`.

Hmm, actually now that I think about this, it really seems like two different functions, doesn't it? And combining them together is leading to this awkward state, where the `get` also `sets`. 

Maybe for querying current sessions we can have `getCurrentSessionAndWindowId` which just returns `this._getSessionId()` and `this._getWindowId()`,

and rename `getSessionAndWindowId` to `getOrCreateSessionAndWindowId` ? Should also make any code that calls getSessionAndWindowId clearer, that this is the function where session increment is happening.

Reply:
Also sort-of okay with just the parameter rename. 

Since this is a feature and not a fix though, I quite like leaving things better than we found them 😅

Reply:
I really like the `readOnly = False` concept. That makes sense to me.

I don't want to split this out into 2 functions because we should pretty much always be "getting and updating" the value. Just reading the value is a pretty rare case, so I don't want to encourage it. If someone needs to do this, they should probably understand the nuance of what they're doing, and I think having to set a parameter encourages that.

Proposal:
* The "public function" here is: `getAndUpdateSessionAndWindowId`
* With the parameter `readOnly`

I think `getAndUpdate...` is accurate because calling it will always update the timestamp associated with the `sessionId`.

Reply:
On seconds thought, maybe it's `checkAndGetSessionAndWindowId`. That feels the least awkward to me...

Reply:
Makes sense!